common/flatpak-dir.c

```diff
@@ -7651,6 +7697,7 @@ flatpak_dir_install (FlatpakDir          *self,
         {
           g_autoptr(FlatpakOciRegistry) registry = NULL;
           g_autoptr(GFile) registry_file = NULL;
+          g_auto(GLnxLockFile) child_repo_lock = { 0, };
```

Comment:
You can't just move the lock around! That changes which areas are locked. In particular, we no longer lock the repo during the actual operation on the child repo that the lock is supposed to protect!